File: android/app/src/main/java/jagtheora/vorbis/VorbisInfo.java
package jagtheora.vorbis;

import jagtheora.misc.SimplePeer;
import jagtheora.ogg.OggPacket;

public final class VorbisInfo extends SimplePeer {

    public int rate;

    public int channels;

    static {
        initFields();
    }

    public static void initFields() {}

    public VorbisInfo() {
        this.init();
        if (this.b()) {
            throw new IllegalStateException();
        }
    }

    protected void clear() {}

    public int headerIn(VorbisComment arg0, OggPacket arg1) { return 0; }

    protected void init() {}
}

File: src/jvmMain/kotlin/jagtheora/vorbis/VorbisInfo.kt
package jagtheora.vorbis

import jagtheora.misc.SimplePeer
import jagtheora.ogg.OggPacket

actual class VorbisInfo : SimplePeer() {

    actual var rate: Int = 0


    actual var channels: Int = 0

    init {
        init()
        check(!this.b())
    }

    actual external fun headerIn(arg0: VorbisComment?, arg1: OggPacket?): Int

    actual external override fun clear()

    actual companion object {
        init {
            initFields()
        }

        /*private*/ actual external fun initFields()

        /*protected*/ actual external fun init()
    }
}
